lib/sync/storage-manager-listener.js

```diff
@@ -0,0 +1,38 @@
+let cachedStore;
+let cachedListener;
+
+module.exports = function getStorageManagerListener(store) {
+  if (cachedListener && (!store || store === cachedStore)) { return cachedListener; }
+
+  if (!store || (typeof store.dispatch !== 'function')) { throw new Error('You must pass in a redux store the first time you call getStorageManagerListener'); }
+
+  // we need to be able to reset the store with one that is passed in
+  // so that we can reset the store between it blocks
+  cachedStore = store;
+  cachedListener = function storageManagerListener(eventName, entity) {
+    if (eventName.indexOf(':') > -1) return;
+
+    // Heuristic for the time being
+    if (entity !== undefined && entity.attributes) {
+      cachedStore.dispatch({
```

Comment:
Food for thought: We can consider making this a getter method so that we can actually just change which store we want to use. What we have now is probably fine, but a pair of methods would give us finer + explicit control over who + how the store changes in test and app code.